### ISSUE TITLE
Don't redeclare infix operators in files

### ISSUE THREAD

**Issue by laurenchavez212** [reazen/relude#205]
When opening Option and opening Relude.Globals in a file, the Infix operators are declared in both places. This causes the open statement to shadow the declaration in Global, giving a warning. 

```ocaml
open Relude.Globals;
let foo = Option.(map(x => x + 1) >> map(Int.toString));
```

**Comment by andywhite37** [comment]
Thanks for the issue report!

I think the easiest fix for this might be for the `Relude` files to use an `open` to get these types of commonly-used operators like `Function` `>>` and `<<`.  We don't tend to use all the other operators in the relude library code b/c of the issues with ad hoc polymorphism.

@mlms13 - do you think it would work if we replaced these lines:

https://github.com/reazen/relude/blob/master/src/option/Relude_Option_Instances.re#L1-L2

With

```reasonml
open Function.Infix;
```

https://github.com/reazen/relude/blob/master/src/Relude_Function.re#L373-L401

That would only fix shadowed uses of the compose operators, but I suspect those are the main culprits.

**Comment by mlms13** [comment]
Yes, there are also some in `IO`, I believe, along with module aliases that might conflict of you open both `Relude.Globals` and `IO` in the same context. In general, we should probably go through some modules and clean up module and function aliases and prefer local opens instead.

**Comment by andywhite37** [comment]
Yeah, agreed.  Would it be terrible to just use a global open for `open Relude_Function.Infix` in the relude code?  It would be kind of painful to do a local open for those compose operators everywhere.

**Comment by mlms13** [comment]
If it works, that seems like it would be fine, and it would avoid polluting the module interfaces.

**Comment by andywhite37** [comment]
I'll take a stab at this, it should be pretty easy

**Comment by andywhite37** [comment]
I'll make a release for this later tonight

**Comment by andywhite37** [comment]
Released in v0.42.0
